RSCore/NSView+RSCore.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  NSView+Extensions.swift
+//  RSCore
+//
+//  Created by Maurice Parker on 11/12/19.
+//  Copyright © 2019 Ranchero Software, LLC. All rights reserved.
+//
+
+import Foundation
+
+extension NSView {
+	
+    public func asImage() -> NSImage {
+		let rep = bitmapImageRepForCachingDisplay(in: bounds)!
+		cacheDisplay(in: bounds, to: rep)
+		
+		let img = NSImage(size: bounds.size)
+		img.addRepresentation(rep)
+		return img
+    }
+	
+}
+
+public extension NSView {
+
+	/**
+	Keeps subview at same size as receiver.
+
+	- Parameter subview: The subview to constrain. Must be a descendant of `self`.
+	*/
+	func addFullSizeConstraints(forSubview subview: NSView) {
+		let d = ["subview": subview]
+
+		var constraints = NSLayoutConstraint.constraints(withVisualFormat: "|[subview|", options: [], metrics: nil, views: d)
+		constraints.append(contentsOf: NSLayoutConstraint.constraints(withVisualFormat: "V:|[subview|", options: [], metrics: nil, views: d))
+
+		self.addConstraints(constraints)
+	}
```

Comment:
Is there an advantage or difference of the code above to the following code?
```
NSLayoutConstraint.activate([
	subview.leadingAnchor.constraint(equalTo: leadingAnchor),
	subview.topAnchor.constraint(equalTo: topAnchor),
	subview.trailingAnchor.constraint(equalTo: trailingAnchor),
	subview.bottomAnchor.constraint(equalTo: bottomAnchor)
])
```
If there is no difference, I suggest using anchors in this case better expresses intent (full sizing is achieved by the equality of anchors), code concision and resilience (compile time checking of anchors vs visual format correctness, fragile identifiers) and more efficient (no string parsing and array resizing).

If I was to guess at one advantage the visual format can give is that the system can create a comprehensive set of constraints for the expressed layout, constraints which layout system could use to optimise layout.

Reply:
Nope, I was just porting what was in the old Objective-C version directly. I actually meant to switch to anchors and then promptly forgot.

Reply:
I like @kielgillard’s suggestion — let’s switch to anchors here.

Reply:
Agreed; I’ll make the change once I get back to my desk.